rust/datafusion/src/physical_plan/hash_utils.rs

```diff
@@ -29,7 +29,7 @@ pub enum JoinType {
 }
 
 /// The on clause of the join, as vector of (left, right) columns.
-pub type JoinOn<'a> = [(&'a str, &'a str)];
+pub type JoinOn = [(String, String)];
```

Comment:
I ran into ownership issues that I couldn't figure out. I am happy to change this back if someone can show me how.

Reply:
This is too small compared to the others ops to justify the effort at this point, IMO.